pkg/pillar/devicenetwork/devicenetwork.go

```diff
@@ -118,28 +130,31 @@ func VerifyDeviceNetworkStatus(status types.DeviceNetworkStatus,
 		if err != nil {
 			errStr := "Onboarding certificate cannot be found"
 			log.Infof("VerifyDeviceNetworkStatus: %s\n", errStr)
-			return false, errors.New(errStr)
+			return false, intfErrMap, errors.New(errStr)
 		}
 		clientCert := &onboardingCert
 		tlsConfig, err = zedcloud.GetTlsConfig(zedcloudCtx.DeviceNetworkStatus,
 			serverName, clientCert, &zedcloudCtx)
 		if err != nil {
-			errStr := fmt.Sprintf("TLS configuration for talking to Zedcloud cannot be found: %s", err)
-
-			log.Infof("VerifyDeviceNetworkStatus: %s\n", errStr)
-			return false, errors.New(errStr)
+			err = fmt.Errorf("TLS configuration for talking to Zedcloud cannot be found: %s", err)
+			log.Infof("VerifyDeviceNetworkStatus: %s\n", err)
+			return false, intfErrMap, err
 		}
 	}
 	zedcloudCtx.TlsConfig = tlsConfig
 	for ix := range status.Ports {
 		err = CheckAndGetNetworkProxy(&status, &status.Ports[ix])
 		if err != nil {
-			errStr := fmt.Sprintf("GetNetworkProxy failed %s", err)
-			log.Errorf("VerifyDeviceNetworkStatus: %s\n", errStr)
-			return false, errors.New(errStr)
+			ifName := status.Ports[ix].IfName
+			err = fmt.Errorf("ifName: %s. Failed to get NetworkProxy. Err:%s",
+				ifName, err)
+			log.Errorf("VerifyDeviceNetworkStatus: %s", err)
+			intfErrMap.AddError(ifName, err, errSrc)
```

Comment:
I asked a question about this. Did you test it? Or just change the code?? I should have kept my mouth shut.

Reply:
I did not test the proxy part of the code. From reading the code - it does the following:

1) If Proxy config is enabled - it checks to see if it can get the config
     - If not - returns an error

2) If there are two mgmt ports with Proxy config - and the first one always fails, but the second one passes - the previous one never passes the test - always errors out.. ( Reading from the code ).
    - Obviously - that part was never tested.
     - I assume in this case, we want to use which ever interface is working..

So - I did not test the real Proxy changes - only with hacked up code ( Forced the code to return an error )..

Let me put the behavior back to return an error.